packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -53,43 +47,71 @@ export const styles = {
   focusVisible: {},
 };
 
+/**
+ * https://github.com/facebook/react/issues/14099#issuecomment-440013892
+ *
+ * @param {function} fn
+ */
+function useEventCallback(fn) {
+  const ref = React.useRef(fn);
+  const useEnhancedEffect = typeof window === 'undefined' ? React.useEffect : React.useLayoutEffect;
+  useEnhancedEffect(() => {
+    ref.current = fn;
+  });
+  return React.useCallback(event => (0, ref.current)(event), []);
+}
+
 /**
  * `ButtonBase` contains as few styles as possible.
  * It aims to be a simple building block for creating a button.
  * It contains a load of style reset and some focus/ripple logic.
  */
-class ButtonBase extends React.Component {
-  state = {};
-
-  buttonRef = React.createRef();
-
-  handleMouseDown = createRippleHandler(this, 'MouseDown', 'start');
-
-  handleMouseUp = createRippleHandler(this, 'MouseUp', 'stop');
-
-  handleMouseLeave = createRippleHandler(this, 'MouseLeave', 'stop', event => {
-    if (this.state.focusVisible) {
-      event.preventDefault();
-    }
-  });
-
-  handleTouchStart = createRippleHandler(this, 'TouchStart', 'start');
-
-  handleTouchEnd = createRippleHandler(this, 'TouchEnd', 'stop');
-
-  handleTouchMove = createRippleHandler(this, 'TouchMove', 'stop');
-
-  handleContextMenu = createRippleHandler(this, 'ContextMenu', 'stop');
```

Comment:
This was dead code.

Reply:
#13664 seems to have removed this incorrectly. There's a regression where the ripple isn't removed if you spam click your left and right mouse button